control-plane-operator/controllers/hostedcontrolplane/cvo/reconcile.go

```diff
@@ -332,6 +340,10 @@ func buildCVOContainerMain(image string) func(c *corev1.Container) {
 			fmt.Sprintf("--listen=0.0.0.0:%d", port),
 			fmt.Sprintf("--serving-cert-file=%s", cpath(cvoVolumeServerCert().Name, corev1.TLSCertKey)),
 			fmt.Sprintf("--serving-key-file=%s", cpath(cvoVolumeServerCert().Name, corev1.TLSPrivateKeyKey)),
+			fmt.Sprintf("--prometheus-query-ca-bundle-file=%s", cpath(cvoVolumeServiceCA().Name, "service-ca.crt")),
+			fmt.Sprintf("--prometheus-query-namespace=%s", metricsQueryNamespace),
```

Comment:
It's not a fair assumption that this would be always available.
What happens if it's not?

Reply:
Hello, Alberto! Thank you for the information and the quick response! I am going through the management cluster capabilities, code, and documentation. I may need to contact my team, also. I'll get back to you soon.

Regarding the goals. There were API changes in the HyperShift that added things like `channel`, `availableUpdates` and `conditionalUpdates` https://github.com/openshift/hypershift/pull/1954. The goal is to fully utilize these fields in the same way as they are utilized in the OpenShift.


These conditional updates are conditional because they have some risks associated with them. These risks **can** be represented internally as PromQL queries. These risks **may** affect the current cluster. These risks/queries then need to be evaluated to decide whether the risks apply to the specific cluster. In the standalone OpenShift, the evaluation is done by the CVO. The CVO runs the conditional update's PromQL query to the `thanos-querier` service in the `openshift-monitoring` namespace. The CVO looks at the result. If the risk doesn't apply to the specific cluster, the CVO then represents the conditional update as an available update because there is no risk of updating to the specific release. The whole concept is detailedly specified in its enhancement ([enhancements/update/targeted-update-edge-blocking.md](https://github.com/openshift/enhancements/blob/master/enhancements/update/targeted-update-edge-blocking.md)).

Reply:
I see, thanks! so I'm re-reading my comment above and It's not accurate. The CVO runs management side and operates the guest cluster where thanos-querier will be in openshift-monitoring. However HCs cluster monitoring does not necessarily guarantees the same metrics than regular ocp, so the conditions/risks might potentially differ or misguide a cvo.status consumer.

So I think we should make revisit enhancements/update/targeted-update-edge-blocking.md and make it sure it either accounts for hypershift or we are aware of the limitations.

Reply:
> However HCs cluster monitoring does not necessarily guarantees the same metrics than regular ocp...

I am sorry for any misunderstanding from my previous message. Yes, after reading a few documentations I have come to this conclusion. This is why this Pull Request is trying to point the CVO in the HCP to an endpoint on the management cluster.

For example, given a management OpenShift cluster, the metrics from the hosted clusters can be stored in the UWM (user workload monitoring) Prometheus in the management cluster. These metrics then can be queried by accessing the `thanos-querier` service in the `openshift-monitoring` namespace in the management cluster. This Pull Request is trying to point the CVO in the HCP to the query service on the management cluster.

Reply:
>For example, given a management OpenShift cluster, the metrics from the hosted clusters can be stored in the UWM (user workload monitoring) Prometheus in the management cluster. These metrics then can be queried by accessing the thanos-querier service in the openshift-monitoring namespace in the management cluster

ah I see, then my first comment applies, but also the second one, because metrics there are not necessarily the ones available in a cvo condition. Also we don't want to give components access to the management cluster kas. Let's discuss about overall design then get back to the PR
/hold

Reply:
@enxebre we had discussed this @csrwng before. CVO running in the HCP needs access to the metrics to evaluate the conditional risk. cc @wking as well. I will setup a meeting sooner to discus this and move forward.